### DISCUSSION TITLE
Getting a list of all modules defined in the build script

### DISCUSSION THREAD

**Discussion by hmf** [com-lihaoyi/mill#2690] [Q&A] [answered]
Is it possible to get a list of all the modules defined in a `build.sc` using the Mill API within that script? My objective is to collect a  list of top modules from within one of the modules in the build script.

Did some searching in the docos and source but may have missed it. 

TIA

**Comment by lefou** [comment, marked as answer]
You get get the direct module children with [`millModuleDirectChildren`](https://mill-build.com/api/latest/mill/define/Module.html). To get a recursive list, you can use `millInternal.modules`. 

We use this in our own build too, e.g. to collect all example build and render them in the documentation.

https://github.com/com-lihaoyi/mill/blob/c75e29c78cfc3c1e04776978bfc8e5697f8ca1aa/build.sc#L999-L1000
